eth/vm/forks/frontier/state.py

```diff
@@ -128,14 +136,16 @@ def build_computation(self,
                     encode_hex(message.storage_address),
                 )
             else:
+                with self.vm_state.trace(tracer):
+                    computation = self.vm_state.get_computation(
+                        message,
+                        transaction_context,
+                    ).apply_create_message()
+        else:
+            with self.vm_state.trace(tracer):
                 computation = self.vm_state.get_computation(
                     message,
-                    transaction_context,
-                ).apply_create_message()
-        else:
-            computation = self.vm_state.get_computation(
-                message,
-                transaction_context).apply_message()
+                    transaction_context).apply_message()
 
         return computation
```

Comment:
`apply_message()` calls `apply_computation` which throws an exception if the tracer does not exist. This function returns `BaseComputation`'s which have an `apply_message` method but if that message is ever called an exception is thrown. The caller of this method will *probably* never call `apply_message` but this is still a smell.